### PR TITLE
Hotfix/header

### PR DESCRIPTION
**Pull request by heiglandreas**
This PR alters the way the HTTP-Status-code is set. It also replaces all the calls to ```header``` and ```exit``` with calls to ```ApiView::setHeader()```, ```ApiView::setResponseCode()```  and ```return```.

That makes way to easier testing.

### CODE REVIEW

**Comment by exussum12** [comment]
Great change ! Instead of just changing the tests to assertnull could they check the correct status code is being set the assert null test doesn't really gain anything

**Comment by heiglandreas** [comment]
You're right @exussum12 in that they gain anything. But there wasn't anything to gain before as they simply asserted that the constructor was left before the headers were set. Therefore this PR just reinstates the current state. It's a refactoring that doesn't introduce new features.

But with this PR merged it's much easier to test controllers and a further PR can then add some more meaningful tests.

File: src/controllers/EventImagesController.php [hunk 1]
```diff
@@ -94,7 +94,12 @@ public function createImage($request, $db)
         $event_mapper->saveNewImage($event_id, $small_filename, $small_width, $small_height, "small");
 
         $location = $request->base . '/' . $request->version . '/events/' . $event_id;
-        header('Location: ' . $location, null, 201);
+
+        $view = $request->getView();
+        $view->setHeader('Location', $location);
+        $view->setResponseCode(201);
+        return;
```
Comment (exussum12) [inline src/controllers/EventImagesController.php]: These returns are not needed.

File: tests/views/ApiViewTest.php [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+     * @preserveGlobalState disabled
+     */
+    public function testThatHeadersAreSet()
+    {
+        $view = new ApiView();
+
+        $view->setResponseCode(202);
+        header('Foo: bar', false, 201);
+
+        ob_start();
```
Comment (heiglandreas) [inline tests/views/ApiViewTest.php]: Sadly not. As **this** tests exactly that part that outputs the view. And the output is done via ```echo``` so I need to get the output buffer here. This test only checks whether the correct header is returned as that was changed (and not tested before) in the refactoring.

A separate test that checks the correct string value is returned should be added with a separate PR.

**Comment by liam-wiltshire** [comment]
@heiglandreas, there seems to be an issue here somewhere - https://github.com/joindin/joindin-web2/blob/master/app/src/Talk/TalkApi.php#L333 is expecting a 204 respons, however after applying this PR, it's returning a 302, which breaks the frontend....
